### ISSUE TITLE
Check the capability for creating/updating posts before give the option to create a new post in the calendar

### ISSUE THREAD

**Issue by andergmartins** [publishpress/PublishPress#799]
Users should be able to view the calendar but not change anything if they don't have the capability for creating or editing posts.

**Comment by andergmartins** [comment]
​I have tested that adding "pp_view_calendar" allows users to see the calendar, but not necessarily edit or add anything. According to my tests they can only edit if they have the "edit_posts" capability. Unless you have a plugin or custom code changing the default capability we check (using the filter "pp_calendar_create_post_cap" you could modify that specific capability).
​
​

**Comment by andergmartins** [comment]
https://secure.helpscout.net/conversation/1429572600/138989?folderId=2557470

**Comment by andergmartins** [comment]
I have removed the capability for creating posts, but the user still can create posts in the calendar. The post creation is only denied if we remove the Edit Post capability as well.

<img width="1274" alt="Screen Shot 2021-03-05 at 10 23 34" src="https://user-images.githubusercontent.com/407131/110121285-f55fd280-7d9c-11eb-905a-ed3d730c910a.png">

**Comment by andergmartins** [comment]
For now, I suggested the user use this code on the `functions.php` file:

```
add_filter('pp_calendar_create_post_cap', 'customize_pp_calendar_create_post_cap');

function customize_pp_calendar_create_post_cap($capability)
{
    return 'create_posts';
}
```

**Comment by andergmartins** [comment]
We need to review the permissions verification in the calendar. If the user can't edit a post the "edit" button is still displayed. It doesn't work, but the form looks the same for all the posts.

**Comment by andergmartins** [comment]
This was fixed on the new calendar, v3.4.0
